### PR TITLE
(bounty) you no longer need ALL parts deployed to use a modsuit (limbless welcome!)

### PR DESCRIPTION
**Pull request by mc-oofert**
## About The Pull Request

you no longer need all parts deployed to activate a modsuit
in order to use a module it needs its required part deployed, eg magnetic attachment = boots needed
passive effects from a module dont work unless the relevant parts are deployed

video todo

## Why It's Good For The Game

fikou wants this and uhh i guess its better that the legless and the armless are capable of using modsuits
however wearing a modsuit onearmed or onelegged properly still needs a seperate PR

## Changelog
:cl:
balance: you no longer need all parts deployed to activate a modsuit, you can use them limbless
/:cl:

### CODE REVIEW

**Review by Fikou — CHANGES_REQUESTED** [review]
thanks i wanted to do this for a while

File: code/modules/mod/mod_activation.dm [hunk 1]
```diff
@@ -167,6 +162,8 @@
 	mod_link.end_call()
 	to_chat(wearer, span_notice("MODsuit [active ? "shutting down" : "starting up"]."))
 	for(var/obj/item/part as anything in get_parts())
+		if(part.loc == src)
+			continue
 		var/datum/mod_part/part_datum = get_part_datum(part)
 		if(do_after(wearer, activation_step_time, wearer, MOD_ACTIVATION_STEP_FLAGS, extra_checks = CALLBACK(src, PROC_REF(get_wearer)), hidden = TRUE))
```
Comment (Fikou) [inline code/modules/mod/mod_activation.dm:168]: this should be its own proc now and you should run it whenever someone tries to extend/retract a part of the turned on suit. also when turning on the suit the control unit should turn on first now (though when turning off the suit the control unit should still be last)

**Comment by DrAmazing343** [comment]
HOOOOLY SHIT IT'S PEAK

**Comment by Xackii** [comment]
This is not a balance, this is a fix. Creator should have fixed this as soon as they started talking about this problem rather than leaving the problem open for 3 years.

**Comment by mc-oofert** [comment]
> This is not a balance, this is a fix. Creator should have fixed this as soon as they started talking about this problem rather than leaving the problem open for 3 years.

still has balance implications

**Comment by Fikou** [comment]
> This is not a balance, this is a fix. Creator should have fixed this as soon as they started talking about this problem rather than leaving the problem open for 3 years.

this is a balance change also fuck you

**Comment by Xackii** [comment]
> > This is not a balance, this is a fix. Creator should have fixed this as soon as they started talking about this problem rather than leaving the problem open for 3 years.
> 
> this is a balance change also fuck you

In that case, why doesn't mod tell you to fuck off, armless, and doesn't run away from you when you lose your arm? Unfinished.

**Comment by mc-oofert** [comment]
> > > This is not a balance, this is a fix. Creator should have fixed this as soon as they started talking about this problem rather than leaving the problem open for 3 years.
> > 
> > this is a balance change also fuck you
> 
> In that case, why doesn't mod tell you to fuck off, armless, and doesn't run away from you when you lose your arm? Unfinished.

the title is a little misleading because you can use it limbless because deploying parts is now optional 
(still needed for relevant module and pressureproofing)

**Comment by Xackii** [comment]
> > > > This is not a balance, this is a fix. Creator should have fixed this as soon as they started talking about this problem rather than leaving the problem open for 3 years.
> > > 
> > > 
> > > this is a balance change also fuck you
> > 
> > 
> > In that case, why doesn't mod tell you to fuck off, armless, and doesn't run away from you when you lose your arm? Unfinished.
> 
> the title is a little misleading because you can use it limbless because deploying parts is now optional (still needed for relevant module and pressureproofing)

Yes, I like it. I don’t like that you don’t get space proof if you don’t wear all the pieces. It's a little annoying when you're forced to replace almost all of your slots with pieces of a mod. Even if the mod doesn’t give me insuls if I’m not wearing mod gloves, but the fact that it won’t give space proof if you don’t put it on completely, which forces you to completely hide your drip in order to get space proof, is probably the main reason why I’ll take eva instead mod.

**Comment by mc-oofert** [comment]
> Yes, I like it. I don’t like that you don’t get space proof if you don’t wear all the pieces. It's a little annoying when you're forced to replace almost all of your slots with pieces of a mod. Even if the mod doesn’t give me insuls if I’m not wearing mod gloves, but the fact that it won’t give space proof if you don’t put it on completely, which forces you to completely hide your drip in order to get space proof, is probably the main reason why I’ll take eva instead mod.

ok you do you

**Comment by Xackii** [comment]
> > Yes, I like it. I don’t like that you don’t get space proof if you don’t wear all the pieces. It's a little annoying when you're forced to replace almost all of your slots with pieces of a mod. Even if the mod doesn’t give me insuls if I’m not wearing mod gloves, but the fact that it won’t give space proof if you don’t put it on completely, which forces you to completely hide your drip in order to get space proof, is probably the main reason why I’ll take eva instead mod.
> 
> ok you do you

AAAAAAAAAAAAAA!!!!!!

**Comment by Fikou** [comment]
> In that case, why doesn't mod tell you to fuck off, armless, and doesn't run away from you when you lose your arm? Unfinished.

because it wasnt coded to, it was coded to turn off if any part retracts back into the suit

**Comment by Xackii** [comment]
> > In that case, why doesn't mod tell you to fuck off, armless, and doesn't run away from you when you lose your arm? Unfinished.
> 
> because it wasnt coded to, it was coded to turn off if any part retracts back into the suit

In that case, why wasn’t this balance mechanic written about here? 
https://github.com/tgstation/tgstation/pull/59109

**Comment by mc-oofert** [comment]
> > > In that case, why doesn't mod tell you to fuck off, armless, and doesn't run away from you when you lose your arm? Unfinished.
> > 
> > because it wasnt coded to, it was coded to turn off if any part retracts back into the suit
> 
> In that case, why wasn’t this balance mechanic written about here? 
> https://github.com/tgstation/tgstation/pull/59109

I'm sorry to tell you but it seems you have no clue what you're talking about

**Comment by Dawnseer** [comment]
Will this make it so that if you get your arm or leg blown off by whatever that the modsuit won't disable? Will it still cover one leg/arm Or will the still remaining arm/leg become uncovered?

**Comment by mc-oofert** [comment]
> Will this make it so that if you get your arm or leg blown off by whatever that the modsuit won't disable? Will it still cover one leg/arm Or will the still remaining arm/leg become uncovered?

you cant wear gloves or boots on 1 arm yet
however if your arm gets cut off and you can no longer wear gloves that part gets retracted and modsuit will still be on

**Comment by mc-oofert** [comment]
i havent seen an issue report about MODs yet

**Comment by Fikou** [comment]
alright i have time to deal with this now

**Review by Fikou — DISMISSED** [review]
my game wont compile so this for now but more soon

**Pull request merged**